api/models/people.py

```diff
@@ -45,72 +43,106 @@ class PersonalInfo(models.Model):
     registration_bot_chat_id = models.IntegerField(blank=True, null=True)
     chatwoot_conversation_id = models.IntegerField(blank=True, null=True)
 
-    comment = models.TextField(blank=True, null=True)
-
     class Meta:
-        ordering = ("last_name", "first_name")  # TODO this could be used for selection algorithm
+        # there may be no phone or tg username, but matching name and email is good enough reason
+        constraints = [
+            models.UniqueConstraint(
+                fields=["first_name", "last_name", "email"], name="full_name_and_email"
+            )
+        ]
+        ordering = ("last_name", "first_name")
+        verbose_name_plural = "personal info records"
+
+    def __str__(self):
+        return f"{self.full_name} ({self.email})"
 
     @property
     def full_name(self):
         return f"{self.first_name} {self.last_name}"
 
 
-class AgeRange(InternalModelWithName):
+class AgeRange(models.Model):
     """Model for age range.  Students have no exact ages, but age ranges. Teachers' preferences
     and group building algorithms are also based on age ranges.
     """
 
-    # When bot registers a teacher, it gives the teacher names of age groups.  I don't think it
-    # makes sense to make the entire model multilingual just because of this one case.
+    age_from = models.PositiveSmallIntegerField()
+    age_to = models.PositiveSmallIntegerField()
 
-    age_from = models.IntegerField()
-    age_to = models.IntegerField()
+    class Meta:
+        constraints = [models.UniqueConstraint(fields=["age_from", "age_to"], name="age_from_to")]
 
+    def __str__(self):
+        return f"Age {self.age_from} to {self.age_to}"
 
-class Coordinator(models.Model):
-    """Model for a coordinator."""
 
-    # We want to give informative related_name (PersonalInfo.coordinator will be misleading),
-    # so an abstract model with personal_info is out of the question.
+class Person(models.Model):
+    """Abstract model for a coordinator/student/teacher. Stores their common fields and methods."""
+
+    comment = models.TextField(blank=True, null=True)
     personal_info = models.OneToOneField(
         PersonalInfo,
         on_delete=models.CASCADE,
         primary_key=True,
-        related_name="as_coordinator",
+        related_name="as_%(class)s",  # produces `.as_coordinator` etc.
     )
+
+    # Statuses are different for different roles, but this automatic field is common for them all.
+    # Date and time of status change can be tracked in LogEvents but this is a convenient shortcut.
+    status_since = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return (
+            f"{self.personal_info.full_name}. Status: {getattr(self, 'status')} "
```

Comment:
Well, PyCharm complains about unresolved attribute reference :) since this attribute is only present in concrete models. BTW this is kinda bad practice from my side because the parent class is aware of an attribute in its children, but I thought that that is OK in a harmless `__str__` method :)